lib/between_meals/knife.rb

```diff
@@ -75,12 +75,15 @@ def cookbook_upload_all
     end
 
     def berks_cookbook_upload_all
+      unless @berks_config.nil?
+        @berks_config = '--config=' + @berks_config
+      end
       @cookbook_dirs.each do |path|
         cookbooks = Dir["#{path}/*"].select { |o| File.directory?(o) }
         cookbooks.each do |cb|
           @logger.warn("Running berkshelf on cookbook: #{cb}")
-          exec!("cd #{path}/#{cb} && #{@berks} install && #{@berks} upload",
-                @logger)
+          exec!("cd #{cb} && #{@berks} install #{@berks_config} && " +
+            "#{@berks} upload #{@berks_config}", @logger)
```

Comment:
you still need @berks_config defined up in `init` - just don't do the `|| ''` bit